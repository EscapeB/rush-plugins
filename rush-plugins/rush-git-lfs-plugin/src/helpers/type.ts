export type NestedRequired<T> = {
  [P in keyof T]-?: NestedRequired<T[P]>;
};
export type NestedPartial<T> = {
  [P in keyof T]?: NestedPartial<T[P]>;
};
