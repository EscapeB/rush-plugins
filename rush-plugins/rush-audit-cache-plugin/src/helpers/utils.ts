export const durationToString = (duration: number): string => {
  if (duration > 60) {
    const minutes: number = Math.floor(duration / 60);
    const seconds: number = duration % 60.0;

    return `${minutes.toFixed(0)} minute${
      minutes === 1 ? "" : "s"
    } ${seconds.toFixed(1)} seconds`;
  } else {
    return `${duration.toFixed(2)} seconds`;
  }
};
