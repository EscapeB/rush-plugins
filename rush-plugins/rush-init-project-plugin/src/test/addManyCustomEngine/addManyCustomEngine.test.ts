import path from 'path';
import fs from 'fs';
import del from 'del';
import { preparePlop } from '../utils';

import type { NodePlopAPI } from 'node-plop';

describe('addManyCustomEngine', () => {
  let plop: NodePlopAPI;
  const baseFolder: string = path.resolve(__dirname, 'plop-templates');
  const destination: string = path.resolve(__dirname, 'output');
  const plopfilePath: string = path.resolve(__dirname, 'plopfile.js');
  beforeEach(async () => {
    await del(destination);
    plop = preparePlop(plopfilePath);
  });
  it('should add many files', async () => {
    const answer = {
      packageName: 'foo'
    };
    plop.setGenerator('testAddManyCustomEngine', {
      description: '',
      prompts: [],
      actions: [
        {
          type: 'addMany',
          force: true,
          destination,
          base: baseFolder,
          templateFiles: [`**/*`, '!init.config.ts', '!init.config.js'],
          globOptions: {
            cwd: baseFolder,
            dot: true,
            absolute: true
          },
          data: {}
        }
      ]
    });

    await plop.getGenerator('testAddManyCustomEngine').runActions(answer);

    const packageJsonFile: string = path.resolve(destination, 'pkg.json');
    expect(fs.existsSync(packageJsonFile)).toBe(true);
    let packageJson = {} as any;
    expect(() => {
      packageJson = JSON.parse(fs.readFileSync(packageJsonFile, 'utf-8'));
    }).not.toThrow();
    expect(packageJson.name).toBe('foo');

    const indexFile: string = path.resolve(destination, 'index.sample');
    expect(fs.existsSync(indexFile)).toBe(true);
    expect(fs.readFileSync(indexFile, 'utf-8')).toMatchSnapshot();
  });
});
