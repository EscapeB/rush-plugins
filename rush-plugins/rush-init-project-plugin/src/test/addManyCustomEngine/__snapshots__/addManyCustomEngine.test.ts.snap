// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addManyCustomEngine should add many files 1`] = `
"/* eslint-disable */
// prettier-ignore
import React from \\"react\\";

interface ImageProps {
  src: string;
}

const Image = (props: ImageProps): JSX.Element => {
  const { src } = props;
  return (
    <img
      className=\\"class\\"
      src={src}
      style={{ width: \\"100%\\", height: \\"100%\\" }}
    />
  );
};

export default Image;
"
`;
